rust/hyperlane-base/src/types/gcs_storage.rs

```diff
@@ -174,6 +175,15 @@ impl CheckpointSyncer for GcsStorageClient {
         Ok(())
     }
 
+    /// Write the agent metadata to this syncer
+    async fn write_metadata(&self, metadata: &AgentMetadata) -> Result<()> {
```

Comment:
Noticing we could reuse some logic across all `CheckpointSyncer` impls, and across all write operations (`write_checkpoint`, `write_metadata`, `write_announcement`). First the data is serialized with `serde_json`, then it's posted in a syncer-specific way. I'll create a ticket for this in case we ever decide to post additional data

Reply:
Ok. I leave it as is then.